python_modules/libraries/dagster-components/dagster_components/core/dsl_schema.py

```diff
@@ -5,22 +5,19 @@
 from dagster._core.definitions.asset_selection import AssetSelection
 from dagster._core.definitions.asset_spec import AssetSpec, map_asset_specs
 from dagster._core.definitions.assets import AssetsDefinition
-from dagster._core.definitions.declarative_automation.automation_condition import (
-    AutomationCondition,
-)
 from dagster._core.definitions.definitions_class import Definitions
 from dagster._record import replace
 from pydantic import BaseModel, Field
 
-from dagster_components.core.component_rendering import RenderingScope, TemplatedValueResolver
+from dagster_components.core.component_rendering import RenderedModel, TemplatedValueResolver
 
 
 class OpSpecBaseModel(BaseModel):
     name: Optional[str] = None
     tags: Optional[Dict[str, str]] = None
 
 
-class AssetAttributesModel(BaseModel):
+class AssetAttributesModel(RenderedModel):
     key: Optional[str] = None
     deps: Sequence[str] = []
```

Comment:
Proposal (from live discussion):

```python
automation_condition=Rendered[Optional[str], RenderingInfo(scope=..., output_type=AutomationCondition)]
```

* `Rendered` can be `Annotated` for now, but we should have a specialized annotated variant soon.
* Right now, all automation condition functions are available globally or injected at the component level. Leave as outstanding work item to allow for more granular scopes.
* Rendered fields can also have descriptions, by default inherited from docstring